### PR TITLE
Add AMG

### PR DESCRIPTION
**Pull request by anwai98**
- Added the Automatic Mask Generation using Grid Search for FT / Vanilla Models

### CODE REVIEW

File: micro_sam/evaluation/automatic_mask_generation.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import imageio.v2 as imageio
+from elf.evaluation import mean_segmentation_accuracy
+
+from micro_sam.util import precompute_image_embeddings
+from micro_sam.instance_segmentation import AutomaticMaskGenerator
+
+from .inference import get_predictor
+
+
+def remove_bg_as_biggest_element(image, masks):
```
Comment (constantinpape) [inline micro_sam/evaluation/automatic_mask_generation.py:17]: Please remove this and use https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/instance_segmentation.py#L53 instead.

File: micro_sam/evaluation/automatic_mask_generation.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                "SA75": sas[5]
+            }
+            tmp_df = pd.DataFrame([result_dict])
+            net_list.append(tmp_df)
+
+    img_gs_df = pd.concat(net_list)
+    img_gs_df.to_csv(img_save_path, index=False)
+
+
+def per_image_amg(args):
```
Comment (constantinpape) [inline micro_sam/evaluation/automatic_mask_generation.py:67]: This should all be independent of LiveCELL. Please have a look how I separated out things in
- https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/evaluation/inference.py#L306 and
- https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/evaluation/livecell.py#L38

The functionality here should be generic and just take `image_paths` and `gt_paths`. It also shouldn't use `argparse`, but provide a clean functional interface instead.
And you can implement livecell specific functions using the functionality from here in `livecell.py`.

Comment (anwai98) [reply micro_sam/evaluation/automatic_mask_generation.py]: Ahha yes, makes total sense to generalise it already! Cleaned up the functionality as per your previous snippets' expectations!

**Comment by constantinpape** [comment]
Thanks for the changes @anwai98. I will go ahead and merge this and then do some more clean up myself.

**Pull request merged**